src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/CourtService.kt

```diff
@@ -236,6 +241,27 @@ class CourtService(
       )
     }
 
+  private fun validateVideoLinkBookingSpecification(specification: VideoLinkBookingSpecification) {
+    validateVideoLinkAppointment(specification.main, "Main")
+    specification.pre?.let { validateVideoLinkAppointment(it, "Pre") }
+    specification.post?.let { validateVideoLinkAppointment(it, "Post") }
+  }
+
+  private fun validateVideoLinkBookingUpdateSpecification(specification: VideoLinkBookingUpdateSpecification) {
+    validateVideoLinkAppointment(specification.main, "Main")
+    specification.pre?.let { validateVideoLinkAppointment(it, "Pre") }
+    specification.post?.let { validateVideoLinkAppointment(it, "Post") }
+  }
+
+  private fun validateVideoLinkAppointment(appointment: VideoLinkAppointmentSpecification, prefix: String) {
```

Comment:
This reads well and I can follow easily so it's fine.  But I was wondering how it would look if you were to make this an extension function called `VideoLinkAppointmentSpecification.validate` - you might lose a lot of the verbosity?

Reply:
Excellent suggestion. I'll get on with it.